tests/farmer_harvester/test_farmer.py

```diff
@@ -0,0 +1,519 @@
+    name: str
+    current_time: float
+    count: int
+    value: Optional[Union[int, Dict[str, Any]]]
+    expected_result: Any
+
+    def __init__(
+        self,
+        p2_singleton_puzzle_hash: bytes32,
+        name: str,
+        current_time: float,
+        count: int,
+        value: Optional[Union[int, Dict[str, Any]]],
+        expected_result: Any,
+    ):
+        prepared_p2_singleton_puzzle_hash = std_hash(b"11223344")
+        self.pool_states = {
+            prepared_p2_singleton_puzzle_hash: {
+                "p2_singleton_puzzle_hash": prepared_p2_singleton_puzzle_hash.hex(),
+                "xxx_since_start": 1,
+                "xxx_24h": [(1689491043, 1)],
+                "current_difficulty": 1,
+            }
+        }
+        self.p2_singleton_puzzle_hash = p2_singleton_puzzle_hash
+        self.name = name
+        self.current_time = current_time
+        self.count = count
+        self.value = value
+        self.expected_result = expected_result
+
+
+class DummyHarvesterPeer:
+    return_invalid_response: bool
+    peer_node_id: bytes32 = bytes32(token_bytes(32))
+
+    def __init__(self, return_valid_response: bool):
+        self.return_invalid_response = return_valid_response
+
+    async def send_message(self, arg1: Any):
+        pass
+
+    async def call_api(self, arg1: Any, arg2: Any):
+        if self.return_invalid_response:
+            return 0
+        return RespondSignatures(
+            plot_identifier="test",
+            challenge_hash=bytes32(token_bytes(32)),
+            sp_hash=bytes32(token_bytes(32)),
+            local_pk=G1Element(),
+            farmer_pk=G1Element(),
+            message_signatures=[(bytes32.from_bytes(b"ff" * 32)), G2Element()],
+        )
+
+
+@dataclass
+class NewProofOfSpaceCase:
+    difficulty: uint64
+    sub_slot_iters: uint64
+    challenge_hash: bytes32
+    sp_hash: bytes32
+    plot_identifier: str
+    signage_point_index: uint8
+    plot_id: bytes32
+    plot_size: uint8
+    plot_challenge: bytes32
+    plot_public_key: G1Element
+    pool_public_key: Optional[G1Element]
+    pool_contract_puzzle_hash: Optional[bytes32]
+    height: uint32
+    proof: bytes
+    pool_config: PoolWalletConfig
+    pool_difficulty: Optional[uint64]
+    authentication_token_timeout: Optional[uint8]
+    use_invalid_peer_response: bool
+    expected_pool_state: Dict[str, Any]
+    marks: Marks = ()
+
+    # This creates a test case whose proof of space passes plot filter and quality check
+    @classmethod
+    def create_verified_quality_case(
+        cls,
+        difficulty: uint64,
+        sub_slot_iters: uint64,
+        pool_url: str,
+        pool_difficulty: Optional[uint64],
+        authentication_token_timeout: Optional[uint8],
+        use_invalid_peer_response: bool,
+        expected_pool_stats: Dict[str, Any],
+    ):
+        p2_singleton_puzzle_hash = bytes32.fromhex("302e05a1e6af431c22043ae2a9a8f71148c955c372697cb8ab348160976283df")
```

Comment:
There are a lot of magic numbers here. I think it would be good to document what they are and why you chose these numbers.

Also, it seems like it would be appropriate the move these constants from the test parameter to the test itself